### PR TITLE
Added TF OpenAi GPT1 Sequence Classification

### LINKED ISSUE #7623: Implement PyTorch and/or TensorFlow sequence classification architectures for causal language models
# 🚀 Feature request

The architecture `GPT2ForSequenceClassification` was added in #7501 in PyTorch. It would be great to have it in TensorFlow (cf. issues #7622), but it would also be great to have it for other causal models: ~OpenAI GPT~, ~CTRL~ (PR opened @elk-cloner), ~TransfoXL~ (PR opened @spatil6)

Below is a list of items to follow to make sure the integration of such an architecture is complete:

- Implement `XXXForSequenceClassification` in `modeling_xxx.py` or `TFXXXForSequenceClassification` in `modeling_tf_xxx.py
- Test that architecture in `tests/test_modeling_xxx.py` or `tests/test_modeling_tf_xxx.py`
- Add that architecture to `__init__.py` and ` docs/source/model_doc/xxx.rst`.

Taking a look at the code changes in #7501 would be a good start.

A very good first issue to get acquainted with the library and its architectures!

### PR DESCRIPTION
**Pull request by spatil6**
This PR implements Sequence classification for TF OpenAi GPT1 model.
TFOpenAIGPTForSequenceClassification uses the last token in order to do the classification, as other causal models (e.g. Transformer XL ,GPT-2) do.

Fixes #7623 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@LysandreJik @jplu

### CODE REVIEW

**Review by jplu — APPROVED** [review]
Awesome!! Thanks for adding this!

File: src/transformers/models/openai/modeling_tf_openai.py [hunk 1]
```diff
@@ -762,3 +763,154 @@ def call(
             hidden_states=transformer_outputs.hidden_states,
             attentions=transformer_outputs.attentions,
         )
+
+
+@add_start_docstrings(
+    """
+    The GPT2 Model transformer with a sequence classification head on top (linear layer).
```
Comment (LysandreJik) [inline src/transformers/models/openai/modeling_tf_openai.py:770]:
```suggestion
    The OpenAI GPT Model transformer with a sequence classification head on top (linear layer).
```

**Review by LysandreJik — APPROVED** [review]
Great, thanks for working on this! LGTM!

**Comment by spatil6** [comment]
Not sure, why test cases are failingfor 'run_tests_tf'. @LysandreJik . 
Let me know any action require from my side.

**Comment by jplu** [comment]
Can you rebase on master, a fix has been recently merged.

**Comment by LysandreJik** [comment]
The tests have already been fixed on `master`, merging! Thanks a lot @spatil6

**Pull request merged**